src/node_lib.h

```diff
@@ -48,22 +62,27 @@ namespace node::lib {
     bool ProcessEvents();
 
     /*
-    Starts the execution of the Node.js main loop, which processes any events in JavaScript.
+    Starts the execution of the Node.js event loop, which processes any events in JavaScript.
     Additionally, the given callback will be executed once per main loop run.
     *Important*: Call `initialize()` before using this method.
     */
-    bool Run(const RunUserLoop & callback);
+    void RunEventLoop(const RunUserLoop & callback);
 
 
     /*********************************************************
      * Stop Node.js engine
      *********************************************************/
 
     /*
-    Stops the *running* Node.js engine.
+    Sends termination event into event queue and runs event queue, until all events have been handled.
     */
-    bool Terminate();
+    void Terminate();
+
 
+    /*
+    Stops the *running* Node.js engine. Clears all events and puts Node.js into idle.

```

Comment:
Does this halt ("stop") execution of Node.js or clear all events?

Reply:
Terminate is supposed to clear all events and halt the event loop, effectively working like a process.exit() call. An option to only halt the running node.js event loop without clearing the events is a good idea, though, and I think we should add it to the API.